sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubSchemaTransformReadConfiguration.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.pubsub;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+
+/**
+ * Configuration for reading from Pub/Sub.
+ *
+ * <p><b>Internal only:</b> This class is actively being worked on, and it will likely change. We
+ * provide no backwards compatibility guarantees, and it should not be implemented outside the Beam
+ * repository.
+ */
+@SuppressWarnings({
+  "nullness" // TODO(https://issues.apache.org/jira/browse/BEAM-10402)
+})
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class PubsubSchemaTransformReadConfiguration {
+
+  /** Instantiates a {@link PubsubSchemaTransformReadConfiguration.Builder}. */
+  public static Builder builder() {
+    return new AutoValue_PubsubSchemaTransformReadConfiguration.Builder();
+  }
+
+  /** The expected schema of the Pub/Sub message. */
+  public abstract Schema getDataSchema();
+
+  /**
+   * The Pub/Sub topic path to write failures.
+   *
+   * <p>See {@link PubsubIO.PubsubTopic#fromPath(String)} for more details on the format of the dead
```

Comment:
Probably we should refer to PubSubIO.Read.withDeadLetterTopic() instead.

Reply:
Thank you, @chamikaramj .  Is the recommendation here to change the name of the AutoValue getter as withDeadLetterTopic instead of getDeadLetterQueue?